src/Amazon.Lambda.Tools/Commands/DeployFunctionCommand.cs

```diff
@@ -374,6 +388,13 @@ protected override async Task<bool> PerformActionAsync()
                     {
                         throw new LambdaToolsException($"Error creating Lambda function: {e.Message}", LambdaToolsException.LambdaErrorCode.LambdaCreateFunction, e);
                     }
+
+                    if(this.GetBoolValueOrDefault(this.FunctionUrlEnable, LambdaDefinedCommandOptions.ARGUMENT_FUNCTION_URL_ENABLE, false).GetValueOrDefault())
+                    {
+                        var authType = this.GetStringValueOrDefault(this.FunctionUrlAuthType, LambdaDefinedCommandOptions.ARGUMENT_FUNCTION_URL_AUTH, false);
+                        await base.CreateFunctionUrlConfig(createRequest.FunctionName, authType);
```

Comment:
1. Consider adding a log entry prior to performing this action
2. Should CreateFunctionUrlConfig have an exception handler, similar to CreateFunctionAsync on line 384?

Reply:
`CreateFunctionAsync` is an SDK call which is why it has the exception handler. `CreateFunctionUrlConfig` is a method in this library and inside that method it makes an SDK call and has exception handler around it. The idea is any exception that comes from the SDK is "expected" and return back a context error message not just the SDK error message. If the logic inside `CreateFunctionUrlConfig` triggers an exception besides the SDK call then it is unexpected and will get caught but the catch block at the start of the error message that will print more diagnostic information for the unexpect error message.

Reply:
Normally I would put the logging message before the call but since I want to print something the new URL I need the logging message afterwards. I could also put a logging message before hand but that seems extra verbose. If there is an error creating the url the user will get an error message from the exception thrown in `CreateFunctionUrlConfig` that says the function url could not be created.

Reply:
Ok -- when I read this, I had thought that CreateFunctionUrlConfig was an SDK method, so your explanation makes sense to me